exaudfclient/base/python/python3/python_ext_dataframe.cc

```diff
@@ -882,6 +908,54 @@ inline void handleEmitPyInt(
     pyResult.reset(PyObject_CallMethodObjArgs(resultHandler, pyColSetMethods[c].second.get(), pyColSetMethods[c].first.get(), pyValue.get(), NULL));
 }
 
+inline void handleEmitPyFloat(
+        int c, int r,
+        std::vector<PyPtr>& columnArrays,
+        std::vector<std::pair<PyPtr, PyPtr>>& pyColSetMethods,
+        std::vector<ColumnInfo>& colInfo,
+        std::vector<std::pair<std::string, int>>& colTypes,
+        PyObject *resultHandler,
+        PyPtr& pyValue,
+        PyPtr& pyResult,
+        PyPtr& pySetNullMethodName){
+    PyPtr pyFloat(PyList_GetItem(columnArrays[c].get(), r));
+    checkPyPtrIsNull(pyFloat);
+    if (isNoneOrNA(pyFloat.get())) {
+        pyResult.reset(PyObject_CallMethodObjArgs(resultHandler, pySetNullMethodName.get(), pyColSetMethods[c].first.get(), NULL));
+        return;
+    }
+
+    switch (colInfo[c].type) {
+        case SWIGVMContainers::INT64:
+        case SWIGVMContainers::INT32:
+        {
+            double value = PyFloat_AsDouble(pyFloat.get());
+            if (value < 0 && PyErr_Occurred())
+                throw std::runtime_error("F-UDF-CL-SL-PYTHON-1139: emit() PY_FLOAT: PyFloat_AsDouble error");
+            if (npy_isnan(value)) {
+                pyResult.reset(
+                  PyObject_CallMethodObjArgs(resultHandler, pySetNullMethodName.get(), pyColSetMethods[c].first.get(), NULL));
+                return;
+            } 
+            pyValue.reset(PyLong_FromLong(static_cast<int64_t>(value)));
```

Comment:
Could it be a problem that you also convert swig int32 to int64 here?

Reply:
I think, this more or less copied from another function. I think, we handle all int-types the same.